setup.py

```diff
@@ -97,9 +97,18 @@ def build_parser():
     )
     parser.add_argument(
         "--no-lto",
+        dest="lto",
+        default=None,
+        action="store_false",
+        help="Disables Link Time Optimization: faster compile times but worse performance.",
+    )
```

Comment:
Why do we need both `--lto` and `--no-lto`? To the user that might look like there would be a third state, where LTO is neither enabled nor disabled.

Reply:
@mosra Yeah, there is where its' picked automatically by the build_type in the CMake file. I wasn't sure how to take care of handling the cached LTO otherwise.

Reply:
Also, I think of this as a way forcing whether LTO is enabled / disabled or letting the default take over. (Otherwise, you would need to clean the build folder to switch between LTO and non-LTO builds. I also don't want to force subprojects that may have LTO enabled by default for whatever reason (like PyBind) from disabling the LTO at that stage if it's not set.

Reply:
@mosra In other words, the states are force LTO on for all compilation units, force it off, or let the subproject's respective CMake decide (as was the case before we added the flag to force enable it).

Reply:
This makes sense. I suppose these could be added to a mutually exclusive arg group.

Reply:
@aclegg3 In this case, the last positional arg is the one that takes precedence.

Reply:
I see, makes sense, thanks for the clarification.